test/preview/components/embedders/test_gradient_document_embedder.py

```diff
@@ -0,0 +1,150 @@
+        data = component.to_dict()
+        assert data == {
+            "type": "GradientDocumentEmbedder",
+            "init_parameters": {"workspace_id": workspace_id, "model_name": "bge-large"},
+        }
+
+    @pytest.mark.unit
+    def test_warmup(self):
+        embedder = GradientDocumentEmbedder(access_token=access_token, workspace_id=workspace_id)
+        embedder._gradient.get_embeddings_model = MagicMock()
+        embedder.warm_up()
+        embedder._gradient.get_embeddings_model.assert_called_once_with(slug="bge-large")
+
+    @pytest.mark.unit
+    def test_warmup_doesnt_reload(self):
+        embedder = GradientDocumentEmbedder(access_token=access_token, workspace_id=workspace_id)
+        embedder._gradient.get_embeddings_model = MagicMock(default_return_value="fake model")
+        embedder.warm_up()
+        embedder.warm_up()
+        embedder._gradient.get_embeddings_model.assert_called_once_with(slug="bge-large")
+
+    @pytest.mark.unit
+    def test_run_fail_if_not_warmed_up(self):
+        embedder = GradientDocumentEmbedder(access_token=access_token, workspace_id=workspace_id)
+
+        with pytest.raises(RuntimeError, match="warm_up()"):
+            embedder.run(documents=[Document(text=f"document number {i}") for i in range(5)])
+
+    @pytest.mark.unit
+    def test_run(self):
+        from gradientai.openapi.client.models.generate_embedding_success import GenerateEmbeddingSuccess
+
+        embedder = GradientDocumentEmbedder(access_token=access_token, workspace_id=workspace_id)
+        embedder._embedding_model = NonCallableMagicMock()
+        embedder._embedding_model.generate_embeddings.return_value = GenerateEmbeddingSuccess(
+            embeddings=[{"embedding": np.random.rand(1024).tolist(), "index": i} for i in range(5)]
+        )
+
+        documents = [Document(text=f"document number {i}") for i in range(5)]
+
+        result = embedder.run(documents=documents)
+
+        assert isinstance(result["documents"], list)
+        assert len(result["documents"]) == len(documents)
+        for doc in result["documents"]:
+            assert isinstance(doc, Document)
+            assert isinstance(doc.embedding, list)
+            assert isinstance(doc.embedding[0], float)
+
+    @pytest.mark.unit
+    def test_run_batch(self):
+        from gradientai.openapi.client.models.generate_embedding_success import GenerateEmbeddingSuccess
+
+        embedder = GradientDocumentEmbedder(access_token=access_token, workspace_id=workspace_id)
+        embedder._embedding_model = NonCallableMagicMock()
+
+        embedder._embedding_model.generate_embeddings.return_value = GenerateEmbeddingSuccess(
+            embeddings=[{"embedding": np.random.rand(1024).tolist(), "index": i} for i in range(110)]
+        )
+
+        documents = [Document(text=f"document number {i}") for i in range(110)]
+
+        result = embedder.run(documents=documents)
```

Comment:
Let's lower the batch size as well to see if setting it has any effect

Reply:
I want to see if you can change the value of `batch` to something like 20, so you'd get more than 2 batches when you pass 110 documents. Not a big deal, just an extra check we can add to this existing test.

Reply:
done!